provider/zone_type_filter.go

```diff
@@ -37,7 +37,26 @@ func NewZoneTypeFilter(zoneType string) ZoneTypeFilter {
 }
 
 // Match checks whether a zone matches the zone type that's filtered for.
-func (f ZoneTypeFilter) Match(zone *route53.HostedZone) bool {
+func (f ZoneTypeFilter) Match(visibility string) bool {
```

Comment:
Would it worth trying out an interface, which we could check by type? It would be somehow nice to have one function instead of separate `Match` functions. I know it also confuses me why AWS always has to be special 😄  but it would be nice trying that one out. WDYT?

Reply:
The only reason is why I bring this up is basically the next provider (e.g. Azure) also needs to implementation and we would have another func.

Reply:
That makes sense. I'll make this change when I get a chance.

Reply:
@njuettner This change has been made